### ISSUE TITLE
The bot is really slow with big datasets...

### ISSUE THREAD

**Issue by AtomynosAtom** [gunthercox/ChatterBot#1939]
I have successfully import a huge data set of Reddit into the bot. It seems to work fine for common questions and is really hilarious at times. But for some user inputs it take forever. Which is honestly not fun to have a conversation with... I wanted to ask if there was any way to decrease the response time with some tweaks or to just limit it to find fewer responses with a deadline (time limit)?

**Comment by AllanNyasha** [comment]
Yes l have also noticed the same thing, l hope a fix is created for that

**Comment by RHQ-Rusty** [comment]
@AtomynosAtom @AllanNyasha 

I'm interested in furthering the development of ChatterBot! I'm using it in a project currently. I let the database train itself ~200 times as a test, and trying to use it renders ChatterBot essentially bricked. Once the DB file reached ~30mb, slowdown starts. As it reaches ~50-60mb, it becomes a crawl. One of my queries, a simple 5 word question, is _still_ processing and it's been upwards of 30 minutes.

Any recommended fixes or findings from you all in regards to processing speed?

**Comment by RHQ-Rusty** [comment]
Additionally - @AtomynosAtom - how were you able to get the Reddit dataset into CB?

**Comment by AtomynosAtom** [comment]
> Additionally - @AtomynosAtom - how were you able to get the Reddit dataset into CB?
I went on reddit to r/datasets and got reddit comments 
https://academictorrents.com/details/85a5bd50e4c365f8df70240ffd4ecc7dec59912b
After that I got some scripts to clean the data and save it as a multiple yaml file. But the thing with yaml file is that it can't start with a symbol and all so I had to replace all those kinds of characters with text and then it imported it to CB many of them failed but I got a small chunk of them.
I had asked the owner about a better way to do this but he didn't have one. 
https://github.com/gunthercox/ChatterBot/issues/2037#issuecomment-691721541

**Comment by noob656** [comment]
Yes i really hope @gunthercox fixes this, its really a pain for having to wait minutes for the chatbot to respond with large datasets.
